### PR TITLE
Added support for loading and saving settings

### PR DESCRIPTION
**Pull request by dkollmann**
This implementation adds two functions:
- nativeSettings.loadSettings(filename, defaultSettings)
- nativeSettings.saveSettings(filename, settings)

So what you do is:
- Load the settings.
- Create the tab. The tab now gets a new argument "savecallback". This callback is called when the current tab is changed or the settings are closed.
- Make your save callback function save the settings to the same file as you loaded them from.

### CODE REVIEW

**Comment by justarandomguyintheinternet** [comment]
Seems like an interesting idea, but the mod is supposed to be really just the UI part. I looked at how you use it in your mod, and while it saves you calling some save to file function in each widgets' callback, you still have to put the value back in the settings table, so yeah could also just call a save to file function in there aswell ;D

**Comment by dkollmann** [comment]
Hey sorry for my late reply. My email settings on github did not allow emails.

The thing is, you need a callback to trigger the save. I think it would actually be cleaner to rephrase this as a "closeTab" callback. Then you can put your save code in it. I also use it to remove additional UI elements I add.

Like here, I add the preview text on top so it does not scroll our of the view. But this means I need a callback to remove it.
https://staticdelivery.nexusmods.com/mods/3333/images/3775/3775-1643832097-1055499494.jpeg

If that would be okay for you, I can clean this up into just the closeTab callback and make a new pull request for that.

Thanks Daniel
